R/init.R

```diff
@@ -0,0 +1,148 @@
+                      lda_alpha=0.100000001,
+                      lda_rho=0.100000001,
+                      lda_D=10000,
+                      lda_epsilon=0.00100000005,
+                      math_mode=0,
+                      minibatch=1)
+    mf_check <- list(rank=0)
+    lrq_check <- list(features="",
+                      lrqdropout=FALSE)
+    stage_poly <- list(sched_exponent = 1.0,
+                       batch_sz = 1000,
+                       batch_sz_no_doubling = TRUE)
+    bootstrap_check <- list(rounds=10,
+                            bs_type="mean")
+    autolink <- list(degree=2)
+    cb <- list(costs=2,
+               cb_type="dr",
+               eval=FALSE)
+    cbify <- list(num_classes=3,
+                  cbify_cs=FALSE,
+                  loss0=0,
+                  loss1=1)
+    nn_check <- list(hidden=3,
+                     inpass=FALSE,
+                     multitask=FALSE,
+                     dropout=FALSE,
+                     meanfield=FALSE)
+    topk <- list(k=3)
+    search <- list(id=0,
+                   search_task=NA_character_,
+                   search_interpolation=NA_character_,
+                   search_rollout=NA_character_,
+                   search_passes_per_policy=1,
+                   search_beta=0.5,
+                   search_alpha=1e-10,
+                   search_total_nb_policies=NA_integer_,
+                   search_trained_nb_policies=NA_integer_,
+                   search_allowed_transitions=NA_character_,
+                   search_subsample_time=NA_real_,
+                   search_neighbor_features=NA_character_,
+                   search_rollout_num_steps=NA_integer_,
+                   search_history_length=1,
+                   search_no_caching=FALSE,
+                   search_xv=FALSE,
+                   search_perturb_oracle=0,
+                   search_linear_ordering=FALSE,
+                   search_active_verify=NA_real_,
+                   search_save_every_k_runs=NA_integer_)
+    boosting_check <- list(num_learners=5,
+                           gamma=0.100000001,
+                           alg="BBM")
+
+    check_lists <- list(general_check=general_check, feature_check=feature_check, optimization_check=optimization_check,
+        sgd_check=sgd_check, bfgs_check=bfgs_check, ftrl_check=ftrl_check, pistol_check=pistol_check, ksvm_check=ksvm_check,
+        OjaNewton_check=OjaNewton_check, svrg_check=svrg_check,
+        binary_check=binary_check, oaa_check=oaa_check, ect_check=ect_check, csoaa_check=csoaa_check, wap_check=wap_check,
+        log_multi=log_multi, lda_check=lda_check, mf_check=mf_check, lrq_check=lrq_check, stage_poly=stage_poly, bootstrap_check=bootstrap_check, autolink=autolink,
+        cb=cb, cbify=cbify,
+        nn_check=nn_check, topk=topk, search=search, boosting_check=boosting_check)
+    flatten_check_lists <- .flatten(check_lists)
+
+    assign("check_lists", check_lists, envir=.pkgenv)
```

Comment:
Let's move this out of `onAttach`, promote it to global status, and enclose inside of an environment. e.g. 

```r
# An environment to hold global variables and settings for rwv.
rwv_global <- new.env(parent = emptyenv())

rwv_global$general_check <- list(random_seed=0, ...)
rwv_global$check_lists <- list(general_check=rwv_global$general_check)
rwv_global$flatten_check_lists <- .flatten(rwv_global$check_lists)
```

Reply:
Yes, I will fix this.

Just for my own better understanding, why don't we want to do it in `onAttach`?

Reply:
Should not really matter.  I tend to do it in `.onAttach()` myself and only have the `new.env()` call outside of it.

Reply:
Strangely, I can't run `.flatten()` (defined in `rhelpers.R`) function outside of `onAttach()`